pyresample/utils.py

```diff
@@ -692,3 +711,411 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
+    elif area_extent is not None or shape is not None:
+        return DynamicAreaDefinition(area_id=area_id, description=description, projection=proj_dict, width=width,
+                                     height=height, area_extent=area_extent, rotation=kwargs.get('rotation'),
+                                     optimize_projection=optimize_projection)
+    raise ValueError('Not enough information provided to create an area definition')
+
+
+def _get_proj_data(projection):
+    """Takes a proj4_dict or proj4_string and returns a proj4_dict and a Proj function."""
+    if isinstance(projection, str):
+        proj_dict = proj4_str_to_dict(projection)
+    elif isinstance(projection, dict):
+        proj_dict = projection
+    else:
+        raise TypeError('Wrong type for projection: {0}. Expected dict or string.'.format(type(projection)))
+    return proj_dict
+
+
+def _convert_rotation(rotation, units):
+    """Convert rotation to degrees."""
+    if isinstance(rotation, DataArray):
+        if hasattr(rotation, 'units'):
+            units = rotation.units
+        if units not in ['deg', 'degrees', 'rad',  'radians']:
+            raise ValueError('units provided to rotation are incorrect: {0}'.format(units))
+        rotation = rotation.data
+    if units in ['rad',  'radians']:
+        rotation *= 180 / math.pi
+    return rotation
+
+
+def _sign(num):
+    """Return the sign of the number provided.
+
+    Returns:
+        1 if number is greater than 0, -1 otherwise
+
+    """
+    return -1 if num < 0 else 1
+
+
+def _round_poles(center, units, p):
+    """Round center to the nearest pole if it is extremely close to said pole.
+
+    Used to work around floating point precision issues .
+
+    """
+    # For a laea projection, this allows for an error of 11 meters around the pole.
+    error = .0001
+    if 'deg' in units:
+        if abs(abs(center[1]) - 90) < error:
+            center = (center[0], _sign(center[1]) * 90)
+    elif 'rad' in units:
+        if abs(abs(center[1]) - math.pi / 2) < error * math.pi / 180:
+            center = (center[0], _sign(center[1]) * math.pi / 2)
+    else:
+        center = p(*center, inverse=True, errcheck=True)
+        if abs(abs(center[1]) - 90) < error:
+            center = (center[0], _sign(center[1]) * 90)
+        center = p(*center, errcheck=True)
+    return center
+
+
+def _distance_from_center_forward(var, center, p, is_radians):
+    """Convert distances in degrees to projection units."""
+    # Interprets radius and resolution as distances between latitudes/longitudes.
+    # Since the distance between longitudes and latitudes is not constant in
+    # most projections, there must be reference point to start from.
+    if center is None:
+        raise ValueError('center must be given to convert radius or resolution from an angle to meters')
+
+    center_as_angle = p(*center, radians=is_radians, inverse=True, errcheck=True)
+    unit_conversion = 90
```

Comment:
could you find a better variable name ?

Reply:
Suggestions? `conversion_factor`?

Reply:
the point was that it is used in a subtraction also, so I felt it was more than just a unit conversion factor

Reply:
Yeah so maybe it should be something like `center_reference`.